sdk/storage/Azure.Storage.Files.DataLake/tests/FileClientTests.cs

```diff
@@ -2780,6 +2940,161 @@ public async Task FlushDataAsync_Error()
                     e => Assert.AreEqual("PathNotFound", e.ErrorCode));
         }
 
+        [RecordedTest]
+        [ServiceVersion(Min = DataLakeClientOptions.ServiceVersion.V2020_08_04)]
+        public async Task FlushDataAsync_Lease_Aquire()
+        {
+            await using DisposingFileSystem test = await GetNewFileSystem();
+
+            // Arrange
+            DataLakeFileClient file = InstrumentClient(test.FileSystem.GetFileClient(GetNewFileName()));
+            await file.CreateIfNotExistsAsync();
+            byte[] data = GetRandomBuffer(Constants.KB);
+
+            using Stream stream = new MemoryStream(data);
+            await file.AppendAsync(stream, 0);
+
+            string leaseId = Recording.Random.NewGuid().ToString();
+            TimeSpan duration = TimeSpan.FromSeconds(15);
+
+            DataLakeFileFlushOptions options = new DataLakeFileFlushOptions
+            {
+                LeaseAction = LeaseAction.Acquire,
+                ProposedLeaseId = leaseId,
+                LeaseDuration = duration
+            };
+
+            // Act
+            await file.FlushAsync(Constants.KB, options);
+
+            // Assert
+            Response<PathProperties> propertiesResponse = await file.GetPropertiesAsync();
+            Assert.AreEqual(DataLakeLeaseStatus.Locked, propertiesResponse.Value.LeaseStatus);
+            Assert.AreEqual(DataLakeLeaseState.Leased, propertiesResponse.Value.LeaseState);
+            Assert.AreEqual(DataLakeLeaseDuration.Fixed, propertiesResponse.Value.LeaseDuration);
+        }
+
+        [RecordedTest]
+        [ServiceVersion(Min = DataLakeClientOptions.ServiceVersion.V2020_08_04)]
+        public async Task FlushDataAsync_Lease_AutoRenew()
+        {
+            await using DisposingFileSystem test = await GetNewFileSystem();
+
+            // Arrange
+            DataLakeFileClient file = InstrumentClient(test.FileSystem.GetFileClient(GetNewFileName()));
+            await file.CreateIfNotExistsAsync();
+            byte[] data = GetRandomBuffer(Constants.KB);
+
+            using Stream stream = new MemoryStream(data);
+
+            string leaseId = Recording.Random.NewGuid().ToString();
+            TimeSpan duration = TimeSpan.FromSeconds(15);
+
+            DataLakeFileAppendOptions appendOptions = new DataLakeFileAppendOptions
+            {
+                LeaseAction = LeaseAction.Acquire,
+                ProposedLeaseId = leaseId,
+                LeaseDuration = duration
+            };
+
+            await file.AppendAsync(stream, 0, appendOptions);
+
+            DataLakeFileFlushOptions options = new DataLakeFileFlushOptions
+            {
+                LeaseAction = LeaseAction.AutoRenew,
+                Conditions = new DataLakeRequestConditions
+                {
+                    LeaseId = leaseId
```

Comment:
This is a little bit weird.  I have chosen to re-use `DataLakeRequestConditions.LeaseId` to specify the `LeaseId` for Flush-based leasing operations, to avoid having a duplicate `LeaseId` property.  It is not necessarily the correct design decision.